src/translate.py

```diff
@@ -4000,16 +4023,12 @@ def address_of_gsym(self, sym_name: str) -> AddressOf:
                 asm_data_entry=self.asm_data_value(sym_name),
             )
 
-        type = Type.ptr(sym.type)
-        if self.typemap:
-            ctype = self.typemap.var_types.get(sym_name)
-            if ctype:
-                ctype_type, dim = ptr_type_from_ctype(ctype, self.typemap)
-                sym.array_dim = dim
+            ctype_type = self.universe.get_var_type(sym_name)
+            if ctype_type is not None:
                 sym.type_in_typemap = True
-                type.unify(ctype_type)
-                type = ctype_type
-        return AddressOf(sym, type=type)
+                sym.type.unify(ctype_type)
+
+        return AddressOf(sym, type=sym.type.reference())
```

Comment:
pre-existing, but it's a bit weird that we decay array to pointer here for the type but not for the value. iirc there is some reason for this, but it doesn't easily come to mind at 2 AM :p, and we should really have some comments that explains the setup for arrays. I'll try to remember it when I next continue reviewing...